src/main/frontend/cli.py

```diff
@@ -413,80 +112,180 @@ def color_scores(scores: list[list[Dependency, float]]) -> \
+def parse_requirements(**kwargs) -> UserRequirements:
     """
-    Executes the command that analyzes an SBOM.
+    Parses the requirements from the arguments.
 
     Args:
-        args (Namespace): The parsed arguments of the program.
+        **kwargs (Any): The arguments.
+
+    Returns:
+        UserRequirements: The parsed user requirements.
     """
-    output, verbose, backend = parse_arguments_shared(args)
-    path, requirements = parse_analyze_arguments(args)
+    code_vulnerabilities: int = kwargs.get("code_vulnerabilities")
+    maintenance: int = kwargs.get("maintenance")
+    continuous_testing: int = kwargs.get("continuous_testing")
+    source_risk_assessment: int = kwargs.get("source_risk_assessment")
+    build_risk_assessment: int = kwargs.get("build_risk_assessment")
+
+    return UserRequirements({
+        RequirementsType.CODE_VULNERABILITIES: code_vulnerabilities,
+        RequirementsType.MAINTENANCE: maintenance,
+        RequirementsType.CONTINUOUS_TESTING: continuous_testing,
+        RequirementsType.SOURCE_RISK_ASSESSMENT: source_risk_assessment,
+        RequirementsType.BUILD_RISK_ASSESSMENT: build_risk_assessment,
+    })
+
+
+def validate_backend(_ctx, _param, value: str):
+    """
+    Validates that the backend URL is a valid URL.
+    """
+    value = value.replace("localhost", "127.0.0.1")
+    validation_res = validators.url(value)
+    if isinstance(validation_res, validators.ValidationError):
+        print(validation_res)
+        raise click.BadParameter("Invalid backend URL.")
+    else:
+        return value
+
+
+def validate_git_token(_ctx, _param, value: str):
+    """
+    Validates that a GitHub Personal Access Token is valid.
+    """
+    url = "https://api.github.com/user"
+    headers = {"Authorization": f"token {value}"}
+    response = requests.get(url, headers=headers, timeout=5)
+
+    # TODO: Could do more extensive error-checking here.
+    #       For example, different messages for different status codes.
+    if response.status_code != 200:
+        raise click.BadParameter("Failed to authenticate token. "
+                                 + f"Status code: {response.status_code}")
+    else:
+        return value
 
-    with open(path, encoding='utf-8') as f:
-        sbom_dict: dict = json.load(f)
-        sbom = Sbom(sbom_dict)
 
-    # print(sbom.to_dict())
-    api = FrontEndAPI(backend)
-    # TODO handle errors
+@click.group(context_settings={"max_content_width": 100})
+def ossqa_cli():
+    """
+    The entry point of the program.
+    """
 
-    scored_sbom: Sbom = api.analyze_sbom(sbom, requirements)
 
-    scores = scored_sbom.dependency_manager.get_scored_dependencies()
+@ossqa_cli.command(help="Analyze the given SBOM.")
+@click.argument("path", required=True,
+                type=click.Path(exists=True, dir_okay=False, path_type=Path))
+@click.option("-g", "--git-token", type=str, envvar="GITHUB_AUTH_TOKEN",
+              required=True, callback=validate_git_token,
```

Comment:
Think that the git token should not be required, since we could use the environment to get the key

Reply:
The git token being required has no effect on whether you need to input it or not, since it defaults to GITHUB_AUTH_TOKEN. I did remove it however, and have written that it defaults to the environment variable.